cloudinit/sources/DataSourceAzure.py

```diff
@@ -405,6 +405,10 @@ def crawl_metadata(self):
         candidates = [self.seed_dir]
         if os.path.isfile(REPROVISION_MARKER_FILE):
             candidates.insert(0, "IMDS")
+            msg = ('Reprovision marker file already exists '
+                   'before crawl of Azure metadata')
+            LOG.warning(msg)
+            report_diagnostic_event(msg)
```

Comment:
It looks like we're logging alongside ~every `report_diagnostic_event` call; I wonder if it might make more sense to change the signature of the function to `report_diagnostic_event(msg, log_func)` and then have, for example, this call look like the below?

```suggestion
            report_diagnostic_event(msg, LOG.warning)
```

Reply:
That's a very good suggestion @OddBloke! I'll implement that in the separate PR that will focus on increased logging/diagnostic reports.

Reply:
Thanks for the suggestion on how to split up the PR. I'll split it up so that 1 is goal state related and the other one is logging/telemetry related.